### PR TITLE
Feature/mc 9440 create a forwarding rule

### PR DESCRIPTION
**Pull request by JamieWulala**

### CODE REVIEW

File: source/includes/gcp/_forwarding_rules.md [hunk 1]
```diff
@@ -1,9 +1,150 @@
 #### Forwarding rules
 
 A forwarding rule and its corresponding IP address represent the frontend configuration of a load balancer.
+
+<!-------------------- LIST FORWARDING RULE -------------------->
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: RULES

File: source/includes/gcp/_forwarding_rules.md [hunk 2]
```diff
@@ -1,9 +1,150 @@
...
+
+Attributes | &nbsp;
+------- | -----------
+`creationTimestamp`<br/>*string* | Creation timestamp in RFC3339 text format.
+`description` <br/>*string* | An optional description of this resource.
+`IPAddress`<br/>*string* | The IP address that this forwarding rule serves. 
+`IPProtocol`<br/>*enum* | The IP protocol to which this rule applies. The default protocol is TCP.
+`ipVersion`<br/>*enum* | The IP version that will be used by this forwarding rule. The default IP version is IPV4.
+`kind`<br/>*string* | Type of the resource.
+`loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. Possible values are INTERNAL and EXTERNAL.
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: sinc we're only doing EXTERNAL load balancer at the moment, not sure we should mention INTERNAL here.

File: source/includes/gcp/_forwarding_rules.md [hunk 3]
```diff
@@ -1,9 +1,150 @@
...
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: see comment above.

File: source/includes/gcp/_forwarding_rules.md (hunk 1 above)
Comment (evanluc) [inline source/includes/gcp/_forwarding_rules.md]:
```suggestion
<!-------------------- LIST FORWARDING RULES -------------------->
```

File: source/includes/gcp/_forwarding_rules.md [hunk 4]
```diff
@@ -1,9 +1,150 @@
...
+</code>
+
+Retrieve a list of all forwarding rules
+
+Attributes | &nbsp;
+------- | -----------
+`creationTimestamp`<br/>*string* | Creation timestamp in RFC3339 text format.
+`description` <br/>*string* | An optional description of this resource.
+`IPAddress`<br/>*string* | The IP address that this forwarding rule serves. 
+`IPProtocol`<br/>*enum* | The IP protocol to which this rule applies. The default protocol is TCP.
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: change enum to string, this is a JSON API, enum only makes sense in the backend.

File: source/includes/gcp/_forwarding_rules.md (hunk 2 above)
Comment (evanluc) [inline source/includes/gcp/_forwarding_rules.md]:
```suggestion
`loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. EXTERNAL is the only value supported.
```

File: source/includes/gcp/_forwarding_rules.md [hunk 5]
```diff
@@ -1,9 +1,150 @@
...
+Required | &nbsp;
+------- | -----------
+`name`<br/>*string* | The display name of the forwarding rule
+`portRange`<br/>*string* | The destination ports that forwarding rule forwards packets with.
+`shortTarget`<br/>*string* | The name of the target resource.
+
+Optional | &nbsp;
+------- | -----------
+`description`<br/>*string* | Description of the forwarding rule.
+`reserveStatic`<br/>*boolean* | Whether the operation reserves a new static IP or not. The default value is false.
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: Can we rename this variable, it's not clear what it is, `reserveStaticIp`

File: source/includes/gcp/_forwarding_rules.md (hunk 3 above)
Comment (evanluc) [inline source/includes/gcp/_forwarding_rules.md]:
```suggestion
`loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. EXTERNAL is the only value supported.
```

File: source/includes/gcp/_forwarding_rules.md (hunk 5 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_forwarding_rules.md]: need a bit more detail here, as the example JSON doesn't show all possible scenarios, which we could do.

If the value is false and if no `shortIPAddress` is provided, an ephemeral external IP address will be assigned. 
If the value is false and `shortIPAddress` is provided, then the existing external IP address will be assigned.
If the value is true, a new external static IP will be reserved and assigned.

File: source/includes/gcp/_forwarding_rules.md [hunk 6]
```diff
@@ -1,9 +1,150 @@
...
+`shortIPAddress`<br/>*string* | The name of a existing IP address.
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: we need a bit more details here.

The name of an existing global external IP to assign to the rule. This argument is only valid in conjunction with `reserveStaticIp` being `false`.

File: source/includes/gcp/_forwarding_rules.md (hunk 1 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: done

File: source/includes/gcp/_forwarding_rules.md (hunk 2 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: change it to "The default value is EXTERNAL"

File: source/includes/gcp/_forwarding_rules.md (hunk 3 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: same as above

File: source/includes/gcp/_forwarding_rules.md (hunk 1 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: done

File: source/includes/gcp/_forwarding_rules.md (hunk 4 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: done

File: source/includes/gcp/_forwarding_rules.md (hunk 2 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: change it to "The default value is EXTERNAL"

File: source/includes/gcp/_forwarding_rules.md (hunk 5 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: done

File: source/includes/gcp/_forwarding_rules.md (hunk 3 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: same as above

File: source/includes/gcp/_forwarding_rules.md (hunk 6 above)
Comment (JamieWulala) [reply source/includes/gcp/_forwarding_rules.md]: done

File: source/includes/gcp/_forwarding_rules.md [hunk 7]
```diff
@@ -1,9 +1,150 @@
...
+```
+
+<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/forwardingrules</code>
+
+Create a new forwarding rule
+
+Required | &nbsp;
+------- | -----------
+`name`<br/>*string* | The display name of the forwarding rule
+`portRange`<br/>*string* | The destination ports that forwarding rule forwards packets with.
```
Comment (SeventhOfNine) [inline source/includes/gcp/_forwarding_rules.md]: `ports` -> `port`, we don't handle a range as such on creation.

Comment (SeventhOfNine) [reply source/includes/gcp/_forwarding_rules.md]: `The port which will be allocated to the forwarding rule. Supported ports are: 80 or 8080 (HTTP) and 443 (HTTPS)`

**Review by evanluc — APPROVED** [review]

**Review by SeventhOfNine — CHANGES_REQUESTED** [review]
Please finish the last few comments.

File: source/includes/gcp/_forwarding_rules.md (hunk 6 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_forwarding_rules.md]: @JamieWulala I don't see any changes done here.

File: source/includes/gcp/_forwarding_rules.md (hunk 7 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_forwarding_rules.md]: @JamieWulala I don't see any changes done here.

**Pull request merged**